examples/machine_translation/train.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+from absl import app
+from absl import flags
+
+from examples.machine_translation.data import get_dataset_and_tokenizer
+from examples.machine_translation.model import TranslationModel
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_integer("num_epochs", 1, "Number of epochs to train.")
+flags.DEFINE_integer("steps_per_epoch", None, "Number of steps per epoch.")
```

Comment:
These are finite sized datasets right? Why do we need this?

Reply:
ohh, this is useful if users do not want to run through the whole datasets. Particularly useful for debugging.